irohad/ordering/impl/on_demand_ordering_gate.cpp

```diff
@@ -25,27 +25,25 @@ OnDemandOrderingGate::OnDemandOrderingGate(
     std::shared_ptr<cache::OrderingGateCache> cache,
     std::shared_ptr<shared_model::interface::UnsafeProposalFactory> factory,
     std::shared_ptr<ametsuchi::TxPresenceCache> tx_cache,
-    consensus::Round initial_round,
     logger::LoggerPtr log)
     : log_(std::move(log)),
       ordering_service_(std::move(ordering_service)),
       network_client_(std::move(network_client)),
       events_subscription_(events.subscribe([this](auto event) {
         // exclusive lock
-        std::unique_lock<std::shared_timed_mutex> lock(mutex_);
+        consensus::Round current_round_;
         visit_in_place(event,
-                       [this](const BlockEvent &block_event) {
+                       [this, &current_round_](const BlockEvent &block_event) {
```

Comment:
Maybe leave todo here about generalization:
```suggestion
                       [this, &current_round_](const auto &event) {
                         log_->debug("{}",  event);
                         current_round_ = event.round;
                       }
```
It could be possible when we generalize block empty events with toString (operator<<) method.